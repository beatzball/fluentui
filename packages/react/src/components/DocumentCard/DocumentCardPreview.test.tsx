import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { DocumentCardPreview } from './DocumentCardPreview';
import { TestImages } from '@fluentui/example-data';

describe('DocumentCardPreview', () => {
  it('renders DocumentCardPreview with an Image (no icon) and "alt" prop', () => {
    const component = renderer.create(
      <DocumentCardPreview
        previewImages={[
          {
            name: 'Contoso Marketing Presentation',
            linkProps: {
              href: 'http://bing.com',
            },
            previewImageSrc: TestImages.documentPreview,
            width: 144,
            alt: 'Some fancy Contoso Image',
          },
        ]}
      />,
    );
    const root = component.root;

    const imgWrapper = root.findAll(n => {
      if (!n.props.className) {
        return false;
      }
      return n.type === 'div' && n.props.className.indexOf('ms-DocumentCardPreview-icon') > -1;
    });
    expect(imgWrapper).toHaveLength(0);

    const img = root.find(n => n.type === 'img');
    expect(img.props.alt).toEqual('Some fancy Contoso Image');

    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders DocumentCardPreview with an Image (w/ icon) and "alt" prop', () => {
    const component = renderer.create(
      <DocumentCardPreview
        previewImages={[
          {
            name: 'Contoso Marketing Presentation',
            linkProps: {
              href: 'http://bing.com',
            },
            iconSrc: TestImages.iconPpt,
            width: 144,
            alt: 'Some fancy Contoso Image',
          },
        ]}
      />,
    );
    const root = component.root;

    const imgWrapper = root.findAll(n => {
      if (!n.props.className) {
        return false;
      }
      return n.type === 'div' && n.props.className.indexOf('ms-DocumentCardPreview-icon') > -1;
    });
    expect(imgWrapper).toHaveLength(1);

    const img = root.findAll(n => {
      return n.type === 'img';
    });
    expect(img[0].props.alt).toEqual('Some fancy Contoso Image');
    expect(img[1].props.alt).toEqual('Some fancy Contoso Image');

    expect(component.toJSON()).toMatchSnapshot();
  });
});
