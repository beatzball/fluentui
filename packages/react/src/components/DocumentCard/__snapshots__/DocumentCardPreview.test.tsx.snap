// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCardPreview renders DocumentCardPreview with an Image (no icon) and "alt" prop 1`] = `
<div
  className=
      ms-DocumentCardPreview
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #faf9f8;
        border-bottom: 1px solid #edebe9;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 12px;
        font-weight: 400;
        overflow: hidden;
        position: relative;
      }
>
  <div>
    <div
      className=
          ms-Image
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            overflow: hidden;
          }
      style={
        Object {
          "height": undefined,
          "width": 144,
        }
      }
    >
      <img
        alt="Some fancy Contoso Image"
        className=
            ms-Image-image
            ms-Image-image--portrait
            is-notLoaded
            is-fadeIn
            {
              display: block;
              height: auto;
              opacity: 0;
              width: 100%;
            }
        onError={[Function]}
        onLoad={[Function]}
        role="presentation"
        src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/document-preview.png"
      />
    </div>
  </div>
</div>
`;

exports[`DocumentCardPreview renders DocumentCardPreview with an Image (w/ icon) and "alt" prop 1`] = `
<div
  className=
      ms-DocumentCardPreview
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #faf9f8;
        border-bottom: 1px solid #edebe9;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 12px;
        font-weight: 400;
        overflow: hidden;
        position: relative;
      }
>
  <div>
    <div
      className=
          ms-Image
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            overflow: hidden;
          }
      style={
        Object {
          "height": undefined,
          "width": 144,
        }
      }
    >
      <img
        alt="Some fancy Contoso Image"
        className=
            ms-Image-image
            ms-Image-image--portrait
            is-notLoaded
            is-fadeIn
            {
              display: block;
              height: auto;
              opacity: 0;
              width: 100%;
            }
        onError={[Function]}
        onLoad={[Function]}
        role="presentation"
      />
    </div>
    <div
      className=
          ms-Image
          ms-DocumentCardPreview-icon
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            bottom: 10px;
            font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
            font-size: 14px;
            font-weight: 400;
            left: 10px;
            overflow: hidden;
            position: absolute;
          }
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
    >
      <img
        alt="Some fancy Contoso Image"
        className=
            ms-Image-image
            ms-Image-image--portrait
            is-notLoaded
            is-fadeIn
            {
              display: block;
              opacity: 0;
            }
        onError={[Function]}
        onLoad={[Function]}
        role="presentation"
        src="https://static2.sharepointonline.com/files/fabric/assets/item-types/32/pptx.png"
      />
    </div>
  </div>
</div>
`;
